### PR TITLE
added put and delete method to controllers

### PR DESCRIPTION
**Pull request by dima-semeniuk**

### CODE REVIEW

File: src/main/java/mate/academy/webapp/repository/BookRepository.java [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
+public interface BookRepository extends JpaRepository<Book, Long> {
 
-    Optional<Book> findById(Long id);
+    @Modifying
+    @Transactional
+    @Query(value = "UPDATE Book b SET b.title = ?1, b.author = ?2, "
+            + "b.isbn = ?3, b.price = ?4, b.description = ?5, "
+            + "b.coverImage = ?6 WHERE b.id = ?7")
+    void updateBookById(String title, String author, String isbn, BigDecimal price,
+                        String description, String coverImage, Long id);
```
Comment (HovorukhaBohdan) [inline src/main/java/mate/academy/webapp/repository/BookRepository.java:18]: Method `save` will update entity, if it exists. But this implementation looks normal to me.

**Review by HovorukhaBohdan — APPROVED** [review]
LGTM

File: src/main/java/mate/academy/webapp/repository/BookRepository.java [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-    Book save(Book book);
+public interface BookRepository extends JpaRepository<Book, Long> {
 
-    Optional<Book> findById(Long id);
+    @Modifying
+    @Transactional
+    @Query(value = "UPDATE Book b SET b.title = ?1, b.author = ?2, "
+            + "b.isbn = ?3, b.price = ?4, b.description = ?5, "
+            + "b.coverImage = ?6 WHERE b.id = ?7")
+    void updateBookById(String title, String author, String isbn, BigDecimal price,
```
Comment (vovasalyha) [inline src/main/java/mate/academy/webapp/repository/BookRepository.java:17]: It's a redundant method. Use one of existing.

File: src/main/java/mate/academy/webapp/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -33,8 +33,26 @@ public BookDto findById(Long id) {
...
     @Override
     public List<BookDto> getAll() {
-        return bookRepository.getAll().stream()
+        return bookRepository.findAll().stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
```
Comment (vovasalyha) [inline src/main/java/mate/academy/webapp/service/impl/BookServiceImpl.java:42]: Is this method atomic? Think about using a transaction.

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include: 
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/db.changelog-master.yaml:3]: Don't forget about empty lines.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments. Also, pay attention to your commit messages. You have one commit with message "added put and delete method to controllers", but you have more other changes (like liquibase support, JPA support etc). Make several purposeful commits with meaningful message.

File: src/main/java/mate/academy/webapp/service/impl/BookServiceImpl.java (hunk 3 above)
Comment (kshuryhin) [reply src/main/java/mate/academy/webapp/service/impl/BookServiceImpl.java:42]: I guess this one is still actual @dima-semeniuk

**Review by kshuryhin — CHANGES_REQUESTED** [review]
Looks good, but see the comments

**Pull request merged**